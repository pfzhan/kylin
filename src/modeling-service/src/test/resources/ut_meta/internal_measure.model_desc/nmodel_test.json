{
  "uuid": null,
  "name": "f",
  "owner": "",
  "project": "tpch",
  "description": "",
  "alias": "f",
  "fact_table": "DEFAULT.TEST_KYLIN_FACT",
  "join_tables": [],
  "simplified_dimensions": [
    {
      "name": "CAL_DT",
      "column": "TEST_KYLIN_FACT.CAL_DT",
      "status": "DIMENSION",
      "datatype": "date"
    }
  ],
  "simplified_measures": [
    {
      "expression": "COUNT",
      "name": "COUNT_ALL",
      "parameter_value": [
        {
          "type": "constant",
          "value": 1,
          "table_guid": null
        }
      ],
      "return_type": ""
    },
    {
      "name": "sum_deal_amount",
      "expression": "SUM",
      "return_type": "",
      "comment": "",
      "parameter_value": [
        {
          "type": "column",
          "value": "TEST_KYLIN_FACT.DEAL_AMOUNT"
        }
      ]
    },
    {
      "name": "corr1",
      "expression": "CORR",
      "comment": "",
      "parameter_value": [
        {
          "type": "column",
          "value": "TEST_KYLIN_FACT.PRICE"
        },
        {
          "type": "column",
          "value": "TEST_KYLIN_FACT.ITEM_COUNT"
        }
      ]
    },
    {
      "name": "corr2",
      "expression": "CORR",
      "comment": "",
      "parameter_value": [
        {
          "type": "column",
          "value": "TEST_KYLIN_FACT.ITEM_COUNT"
        },
        {
          "type": "column",
          "value": "TEST_KYLIN_FACT.SELLER_ID"
        }
      ]
    },
    {
      "name": "corr_cc",
      "expression": "CORR",
      "comment": "",
      "parameter_value": [
        {
          "type": "column",
          "value": "TEST_KYLIN_FACT.DEAL_AMOUNT"
        },
        {
          "type": "column",
          "value": "TEST_KYLIN_FACT.DEAL_YEAR"
        }
      ]
    }
  ],
  "computed_columns": [
    {
      "tableIdentity": "DEFAULT.TEST_KYLIN_FACT",
      "tableAlias": "TEST_KYLIN_FACT",
      "guid": "1638847873207_023499728524260566",
      "columnName": "DEAL_AMOUNT",
      "datatype": "DECIMAL(30,4)",
      "expression": "TEST_KYLIN_FACT.ITEM_COUNT * TEST_KYLIN_FACT.PRICE",
      "table_guid": "1638847855202_0404252953031085"
    },
    {
      "tableIdentity": "DEFAULT.TEST_KYLIN_FACT",
      "tableAlias": "TEST_KYLIN_FACT",
      "guid": "1638847896058_003890294417998175",
      "columnName": "DEAL_YEAR",
      "datatype": "INTEGER",
      "expression": "year(TEST_KYLIN_FACT.CAL_DT)",
      "table_guid": "1638847855202_0404252953031085"
    }
  ],
  "last_modified": 0,
  "filter_condition": "",
  "partition_desc": null,
  "multi_partition_desc": null,
  "management_type": "MODEL_BASED",
  "canvas": {
    "coordinate": {
      "TEST_KYLIN_FACT": {
        "x": 434.49998643663196,
        "y": 257.66665988498266,
        "width": 220,
        "height": 200
      }
    },
    "zoom": 9
  },
  "other_columns": [
    {
      "name": "TRANS_ID",
      "column": "TEST_KYLIN_FACT.TRANS_ID",
      "datatype": "bigint"
    },
    {
      "name": "ORDER_ID",
      "column": "TEST_KYLIN_FACT.ORDER_ID",
      "datatype": "bigint"
    },
    {
      "name": "LSTG_FORMAT_NAME",
      "column": "TEST_KYLIN_FACT.LSTG_FORMAT_NAME",
      "datatype": "varchar(4096)"
    },
    {
      "name": "LEAF_CATEG_ID",
      "column": "TEST_KYLIN_FACT.LEAF_CATEG_ID",
      "datatype": "bigint"
    },
    {
      "name": "LSTG_SITE_ID",
      "column": "TEST_KYLIN_FACT.LSTG_SITE_ID",
      "datatype": "integer"
    },
    {
      "name": "SLR_SEGMENT_CD",
      "column": "TEST_KYLIN_FACT.SLR_SEGMENT_CD",
      "datatype": "smallint"
    },
    {
      "name": "SELLER_ID",
      "column": "TEST_KYLIN_FACT.SELLER_ID",
      "datatype": "integer"
    },
    {
      "name": "PRICE",
      "column": "TEST_KYLIN_FACT.PRICE",
      "datatype": "decimal(19,4)"
    },
    {
      "name": "ITEM_COUNT",
      "column": "TEST_KYLIN_FACT.ITEM_COUNT",
      "datatype": "integer"
    },
    {
      "name": "TEST_COUNT_DISTINCT_BITMAP",
      "column": "TEST_KYLIN_FACT.TEST_COUNT_DISTINCT_BITMAP",
      "datatype": "varchar(4096)"
    },
    {
      "name": "IS_EFFECTUAL",
      "column": "TEST_KYLIN_FACT.IS_EFFECTUAL",
      "datatype": "boolean"
    }
  ],
  "save_only": true,
  "with_base_index": true
}