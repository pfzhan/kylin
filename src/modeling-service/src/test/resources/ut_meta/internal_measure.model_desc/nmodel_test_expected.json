{"uuid":"","last_modified":0,"create_time":0,"version":"4.0.0.0","alias":null,"owner":null,"config_last_modifier":null,"config_last_modified":0,"description":null,"fact_table":null,"fact_table_alias":null,"management_type":"TABLE_ORIENTED","filter_condition":null,"partition_desc":null,"capacity":"MEDIUM","segment_config":{"auto_merge_enabled":null,"auto_merge_time_ranges":null,"volatile_range":null,"retention_range":null,"create_empty_segment_enabled":false},"data_check_desc":null,"semantic_version":0,"storage_type":0,"model_type":null,"all_named_columns":[],"all_measures":[{"name":"COUNT_ALL","function":{"expression":"COUNT","parameters":[{"type":"constant","value":"1"}],"returntype":"bigint"},"column":null,"comment":null,"id":100000,"type":"NORMAL","internal_ids":[]},{"name":"sum_deal_amount","function":{"expression":"SUM","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.DEAL_AMOUNT"}],"returntype":"decimal(38,4)"},"column":null,"comment":"","id":100001,"type":"NORMAL","internal_ids":[]},{"name":"corr1","function":{"expression":"CORR","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.PRICE"},{"type":"column","value":"TEST_KYLIN_FACT.ITEM_COUNT"}],"returntype":"double"},"column":null,"comment":"","id":100002,"type":"EXPANDABLE","internal_ids":[100005,100006,100007,100001,100008]},{"name":"corr2","function":{"expression":"CORR","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.ITEM_COUNT"},{"type":"column","value":"TEST_KYLIN_FACT.SELLER_ID"}],"returntype":"double"},"column":null,"comment":"","id":100003,"type":"EXPANDABLE","internal_ids":[100006,100009,100008,100010,100011]},{"name":"corr_cc","function":{"expression":"CORR","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.DEAL_AMOUNT"},{"type":"column","value":"TEST_KYLIN_FACT.DEAL_YEAR"}],"returntype":"double"},"column":null,"comment":"","id":100004,"type":"EXPANDABLE","internal_ids":[100001,100012,100013,100014,100015]},{"name":"corr1_SUM_TEST_KYLIN_FACT_PRICE_0","function":{"expression":"SUM","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.PRICE"}],"returntype":"decimal(29,4)"},"column":null,"comment":null,"id":100005,"type":"INTERNAL","internal_ids":[]},{"name":"corr1_SUM_TEST_KYLIN_FACT_ITEM_COUNT_1","function":{"expression":"SUM","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.ITEM_COUNT"}],"returntype":"bigint"},"column":null,"comment":null,"id":100006,"type":"INTERNAL","internal_ids":[]},{"name":"corr1_SUM_TEST_KYLIN_FACT_AUTO_CC_0_2","function":{"expression":"SUM","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.AUTO_CC_0"}],"returntype":"decimal(38,7)"},"column":null,"comment":null,"id":100007,"type":"INTERNAL","internal_ids":[]},{"name":"corr1_SUM_TEST_KYLIN_FACT_AUTO_CC_1_3","function":{"expression":"SUM","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.AUTO_CC_1"}],"returntype":"bigint"},"column":null,"comment":null,"id":100008,"type":"INTERNAL","internal_ids":[]},{"name":"corr2_SUM_TEST_KYLIN_FACT_SELLER_ID_4","function":{"expression":"SUM","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.SELLER_ID"}],"returntype":"bigint"},"column":null,"comment":null,"id":100009,"type":"INTERNAL","internal_ids":[]},{"name":"corr2_SUM_TEST_KYLIN_FACT_AUTO_CC_2_5","function":{"expression":"SUM","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.AUTO_CC_2"}],"returntype":"bigint"},"column":null,"comment":null,"id":100010,"type":"INTERNAL","internal_ids":[]},{"name":"corr2_SUM_TEST_KYLIN_FACT_AUTO_CC_3_6","function":{"expression":"SUM","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.AUTO_CC_3"}],"returntype":"bigint"},"column":null,"comment":null,"id":100011,"type":"INTERNAL","internal_ids":[]},{"name":"corr_cc_SUM_TEST_KYLIN_FACT_DEAL_YEAR_7","function":{"expression":"SUM","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.DEAL_YEAR"}],"returntype":"bigint"},"column":null,"comment":null,"id":100012,"type":"INTERNAL","internal_ids":[]},{"name":"corr_cc_SUM_TEST_KYLIN_FACT_AUTO_CC_4_8","function":{"expression":"SUM","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.AUTO_CC_4"}],"returntype":"decimal(38,6)"},"column":null,"comment":null,"id":100013,"type":"INTERNAL","internal_ids":[]},{"name":"corr_cc_SUM_TEST_KYLIN_FACT_AUTO_CC_5_9","function":{"expression":"SUM","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.AUTO_CC_5"}],"returntype":"decimal(38,4)"},"column":null,"comment":null,"id":100014,"type":"INTERNAL","internal_ids":[]},{"name":"corr_cc_SUM_TEST_KYLIN_FACT_AUTO_CC_6_10","function":{"expression":"SUM","parameters":[{"type":"column","value":"TEST_KYLIN_FACT.AUTO_CC_6"}],"returntype":"bigint"},"column":null,"comment":null,"id":100015,"type":"INTERNAL","internal_ids":[]}],"recommendations_count":0,"computed_columns":[{"tableIdentity":"DEFAULT.TEST_KYLIN_FACT","tableAlias":"TEST_KYLIN_FACT","columnName":"DEAL_AMOUNT","expression":"TEST_KYLIN_FACT.ITEM_COUNT * TEST_KYLIN_FACT.PRICE","innerExpression":"`TEST_KYLIN_FACT`.`ITEM_COUNT` * `TEST_KYLIN_FACT`.`PRICE`","datatype":"DECIMAL(30,4)","comment":null,"rec_uuid":null},{"tableIdentity":"DEFAULT.TEST_KYLIN_FACT","tableAlias":"TEST_KYLIN_FACT","columnName":"DEAL_YEAR","expression":"year(TEST_KYLIN_FACT.CAL_DT)","innerExpression":"year(`TEST_KYLIN_FACT`.`CAL_DT`)","datatype":"INTEGER","comment":null,"rec_uuid":null},{"tableIdentity":"DEFAULT.TEST_KYLIN_FACT","tableAlias":"TEST_KYLIN_FACT","columnName":"AUTO_CC_0","expression":"TEST_KYLIN_FACT.PRICE * TEST_KYLIN_FACT.PRICE","innerExpression":"`TEST_KYLIN_FACT`.`PRICE` * `TEST_KYLIN_FACT`.`PRICE`","datatype":"DECIMAL(38,7)","comment":"Auto generated for CORR measure","rec_uuid":null},{"tableIdentity":"DEFAULT.TEST_KYLIN_FACT","tableAlias":"TEST_KYLIN_FACT","columnName":"AUTO_CC_1","expression":"TEST_KYLIN_FACT.ITEM_COUNT * TEST_KYLIN_FACT.ITEM_COUNT","innerExpression":"`TEST_KYLIN_FACT`.`ITEM_COUNT` * `TEST_KYLIN_FACT`.`ITEM_COUNT`","datatype":"INTEGER","comment":"Auto generated for CORR measure","rec_uuid":null},{"tableIdentity":"DEFAULT.TEST_KYLIN_FACT","tableAlias":"TEST_KYLIN_FACT","columnName":"AUTO_CC_2","expression":"TEST_KYLIN_FACT.ITEM_COUNT * TEST_KYLIN_FACT.SELLER_ID","innerExpression":"`TEST_KYLIN_FACT`.`ITEM_COUNT` * `TEST_KYLIN_FACT`.`SELLER_ID`","datatype":"INTEGER","comment":"Auto generated for CORR measure","rec_uuid":null},{"tableIdentity":"DEFAULT.TEST_KYLIN_FACT","tableAlias":"TEST_KYLIN_FACT","columnName":"AUTO_CC_3","expression":"TEST_KYLIN_FACT.SELLER_ID * TEST_KYLIN_FACT.SELLER_ID","innerExpression":"`TEST_KYLIN_FACT`.`SELLER_ID` * `TEST_KYLIN_FACT`.`SELLER_ID`","datatype":"INTEGER","comment":"Auto generated for CORR measure","rec_uuid":null},{"tableIdentity":"DEFAULT.TEST_KYLIN_FACT","tableAlias":"TEST_KYLIN_FACT","columnName":"AUTO_CC_4","expression":"TEST_KYLIN_FACT.DEAL_AMOUNT * TEST_KYLIN_FACT.DEAL_AMOUNT","innerExpression":"(`TEST_KYLIN_FACT`.`ITEM_COUNT` * `TEST_KYLIN_FACT`.`PRICE`) * (`TEST_KYLIN_FACT`.`ITEM_COUNT` * `TEST_KYLIN_FACT`.`PRICE`)","datatype":"DECIMAL(38,6)","comment":"Auto generated for CORR measure","rec_uuid":null},{"tableIdentity":"DEFAULT.TEST_KYLIN_FACT","tableAlias":"TEST_KYLIN_FACT","columnName":"AUTO_CC_5","expression":"TEST_KYLIN_FACT.DEAL_AMOUNT * TEST_KYLIN_FACT.DEAL_YEAR","innerExpression":"(`TEST_KYLIN_FACT`.`ITEM_COUNT` * `TEST_KYLIN_FACT`.`PRICE`) * (year(`TEST_KYLIN_FACT`.`CAL_DT`))","datatype":"DECIMAL(38,4)","comment":"Auto generated for CORR measure","rec_uuid":null},{"tableIdentity":"DEFAULT.TEST_KYLIN_FACT","tableAlias":"TEST_KYLIN_FACT","columnName":"AUTO_CC_6","expression":"TEST_KYLIN_FACT.DEAL_YEAR * TEST_KYLIN_FACT.DEAL_YEAR","innerExpression":"(year(`TEST_KYLIN_FACT`.`CAL_DT`)) * (year(`TEST_KYLIN_FACT`.`CAL_DT`))","datatype":"INTEGER","comment":"Auto generated for CORR measure","rec_uuid":null}],"multi_partition_desc":null,"multi_partition_key_mapping":null,"fusion_id":null}